src/blurry_classifier_module.py

```diff
@@ -0,0 +1,210 @@
+import asyncio
+from typing import ClassVar, List, Mapping, Optional, Sequence
+from typing_extensions import Self
+
+from viam.components.camera import Camera
+from viam.media.video import CameraMimeType, ViamImage
+from viam.module.module import Module
+from viam.proto.app.robot import ComponentConfig
+from viam.proto.common import PointCloudObject, ResourceName
+from viam.proto.service.vision import Classification, Detection
+from viam.resource.base import ResourceBase
+from viam.resource.easy_resource import EasyResource
+from viam.resource.types import Model, ModelFamily
+from viam.services.vision import Vision, CaptureAllResult
+from viam.utils import ValueTypes
+
+import cv2 as cv
+
+from src.utils import decode_image
+
+
+class BlurryDetector(Vision, EasyResource):
+    """
+    BlurryDetector implements a vision service that only supports classifications.
+
+    It inherits from the built-in resource subtype Vision and conforms to the
+    ``Reconfigurable`` protocol, which signifies that this component can be
+    reconfigured. Additionally, it specifies a constructor function
+    ``BlurryDetector.new`` which confirms to the
+    ``resource.types.ResourceCreator`` type required for all models.
+    """
+
+    MODEL: ClassVar[Model] = Model(
+        ModelFamily("viam", "blurry-detector"), "blurry-detector"
+    )
+
+    def __init__(self, name: str):
+        super().__init__(name=name)
+        self.camera = None
+        self.camera_name = None
+        self.blurry_threshold = 1000.0
+
+    @classmethod
+    def new(
+        cls,
+        config: ComponentConfig,
+        dependencies: Mapping[ResourceName, ResourceBase],
+    ) -> Self:
+        """
+        This method creates a new instance of this Vision service.  The default
+        implementation sets the name from the `config` parameter and then calls
+        `reconfigure`.
+
+        Args:
+            config (ComponentConfig): The configuration for this resource
+            dependencies (Mapping[ResourceName, ResourceBase]): The
+                dependencies (both implicit and explicit)
+
+        Returns:
+            Self: The resource
+        """
+        service = cls(config.name)
+        service.reconfigure(config, dependencies)
+        return service
+
+    @classmethod
+    def validate_config(cls, config: ComponentConfig) -> Sequence[str]:
```

Comment:
A note for myself: it feels like line 66 should be `@staticmethod` and this line should have the `cls` argument removed. That gets auto-added if you generate your skeleton from the CLI, as a way to demonstrate to people that this kind of thing is possible. It's not worth the hassle to change every time; just leave it be.